### PR TITLE
VIMC-3857: Add endpoint to return information about unmerged git branches

### PR DESCRIPTION
**Pull request by r-ash**
Corresponding orderly PR https://github.com/vimc/orderly/pull/240

@EmmaLRussell What format do you want to the date in the response? At the moment it is just the string returned by git e.g. "Tue Jun 2 17:15:16 2020 +0100" but we can do some reformatting as you need.

### CODE REVIEW

**Comment by EmmaLRussell** [comment]
We want to order by datetime so I guess something utc-like would be ideal if available. 

> @EmmaLRussell What format do you want to the date in the response? At the moment it is just the string returned by git e.g. "Tue Jun 2 17:15:16 2020 +0100" but we can do some reformatting as you need.

**Comment by r-ash** [comment]
> We want to order by datetime so I guess something utc-like would be ideal if available.
> 
> > @EmmaLRussell What format do you want to the date in the response? At the moment it is just the string returned by git e.g. "Tue Jun 2 17:15:16 2020 +0100" but we can do some reformatting as you need.

Cool. That reminds me of another point I forgot to raise. The order that this is returning here is master first then the other branches in order of last commit (most recent first). I can modify that here fairly easily. I assume json preserves the order of an array?

**Comment by EmmaLRussell** [comment]
I was expecting to pass the branch as a parameter, and just get back the commits for that branch, as per https://docs.google.com/presentation/d/1NhkexrPM0jt7xS4cEiR_cwE7xreUDISgT31ks44zne4/edit#slide=id.g77b361aa14_1_110
So the user would first select a branch and then select a commit in that branch (usually the latest one). Which I was imagining to be two separate calls to orderly.  But if everything's coming back in one, that's OK, we'll just filter in OW.
That branch order sounds perfect. 

> Cool. That reminds me of another point I forgot to raise. The order that this is returning here is master first then the other branches in order of last commit (most recent first). I can modify that here fairly easily.

**Comment by EmmaLRussell** [comment]
Sorry, just posted and deleted a comment that didn't make sense. Yes, that branch order sounds perfect!

> 
> Cool. That reminds me of another point I forgot to raise. The order that this is returning here is master first then the other branches in order of last commit (most recent first). I can modify that here fairly easily.

File: R/api.R [hunk 1]
```diff
@@ -98,6 +99,18 @@ endpoint_git_pull <- function(runner) {
     returning = returning_json("GitPull.schema"))
 }
 
+target_git_branches <- function(runner) {
+  browser()
```
Comment (richfitz) [inline R/api.R:103]: Stray browser here 😀

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM vimc/orderly:master
+FROM vimc/orderly:vimc-3857
```
Comment (richfitz) [inline docker/Dockerfile:1]: The orderly one looks good so let's update this as soon as that's merged

File: tests/testthat/helper-orderly.server.R [hunk 3]
```diff
@@ -51,18 +51,6 @@ wait_for_finished_runner <- function(runner, key) {
 }
 
 
-test_runner <- function(path = tempfile()) {
```
Comment (richfitz) [inline tests/testthat/helper-orderly.server.R:54]: I thought I'd deleted this on 76f8197808512c0fa3ddf734a07118385964801c - where are we up to with integration?

File: tests/testthat/test-api-endpoints.R [hunk 4]
```diff
@@ -577,3 +577,18 @@ test_that("run metadata can get name from config", {
     changelog_types = NULL
   ))
 })
+
+test_that("git branches endpoint", {
+  path <- orderly_prepare_orderly_git_example()
+  branch_data <- data.frame(
+    name = c("master", "dev-branch"),
+    last_commit = c("Mon Jun 1 16:00:41 2020 +0100",
```
Comment (richfitz) [inline tests/testthat/test-api-endpoints.R:585]: Can we make this `last_commit_time` and also add `last_commit_age` which would be an integer in seconds? That will avoid a lot of time zone confusion.

Also this time should be ISO formatted

**Pull request merged**